rangeserver/src/for_testing/in_memory_wal.rs

```diff
@@ -38,13 +36,18 @@ pub struct InMemIterator<'a> {
 impl<'a> Iterator<'a> for InMemIterator<'a> {
     async fn next_offset(&self) -> Result<u64, Error> {
         let wal = self.wal.state.lock().await;
-        let offset = wal.first_offset + self.index;
-        Ok(offset)
+        match wal.first_offset {
+            None => Ok(0),
+            Some(first_offset) => {
+                let offset = first_offset + self.index;
+                Ok(offset)
+            }
+        }
     }
 
     async fn next(&mut self) -> Option<LogEntry<'_>> {
         let wal = self.wal.state.lock().await;
-        let ind = (wal.first_offset + self.index) as usize;
+        let ind = (wal.first_offset.unwrap() + self.index) as usize;
```

Comment:
This `unwrap` is unsafe, right? Should this move after the length check, perhaps?

Reply:
Updated, but I suspect this module is not very correct. I'll improve its safety when I implement logging tests in the RM.